uamqp/session.py

```diff
@@ -214,13 +215,61 @@ def _outgoing_transfer(self, delivery):
         if self.remote_incoming_window <= 0:
             delivery.transfer_state = SessionTransferState.Busy
         else:
+            payload = delivery.frame['payload']
+            payload_size = len(payload)
+
             delivery.frame['delivery_id'] = self.next_outgoing_id
-            self._connection._process_outgoing_frame(self.channel, TransferFrame(**delivery.frame))
+            # calculate the transfer frame encoding size excluding the payload
+            delivery.frame['payload'] = b""
+            encoded_frame = encode_frame(TransferFrame(**delivery.frame))[1]
+            transfer_overhead_size = len(encoded_frame)
+
+            # available size for payload per frame is calculated as following:
+            # remote max frame size - transfer overhead (calculated) - header (8 bytes)
+            available_frame_size = self._connection._remote_max_frame_size - transfer_overhead_size - 8
+
+            start_idx = 0
+            remaining_payload_cnt = payload_size
+            # encode n-1 frames if payload_size > available_frame_size
+            while remaining_payload_cnt > available_frame_size:
```

Comment:
assuming sending large messages will be the most common scenarios (batching in EH/SB), so treating this as the default behavior.